File: app/src/main/java/com/adisa/diningplus/FollowDialogPreference.java
package com.adisa.diningplus;

import android.app.AlertDialog;
import android.content.Context;
import android.content.SharedPreferences;
import android.preference.DialogPreference;
import android.preference.PreferenceManager;
import android.util.AttributeSet;
import android.view.View;
import android.widget.ListView;

import com.adisa.diningplus.adapters.FollowDialogAdapter;

import java.util.HashSet;

/**
 * Created by Adisa on 5/4/2017.
 */

public class FollowDialogPreference extends DialogPreference {
    public FollowDialogPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
        setDialogLayoutResource(R.layout.follow_dialog);
    }

    @Override
    public void onBindDialogView(View view) {
        ListView listView = (ListView) view.findViewById(R.id.followList);
        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getContext());
        HashSet<String> stringSet = (HashSet<String>) preferences.getStringSet("followedItems", new HashSet<String>());
        FollowDialogAdapter followDialogAdapter = new FollowDialogAdapter(getContext(), stringSet);
        listView.setAdapter(followDialogAdapter);

        super.onBindDialogView(view);
    }

    @Override
    protected void onPrepareDialogBuilder(AlertDialog.Builder builder) {
        builder.setNegativeButton(null, null);
        super.onPrepareDialogBuilder(builder);
    }

    @Override
    protected void onDialogClosed(boolean positiveResult) {
        super.onDialogClosed(positiveResult);
        persistBoolean(positiveResult);
    }
}

File: app/src/main/java/com/adisa/diningplus/FollowDialogPreference.kt
package com.adisa.diningplus

import android.app.AlertDialog
import android.content.Context
import android.preference.DialogPreference
import android.preference.PreferenceManager
import android.util.AttributeSet
import android.view.View
import android.widget.ListView
import com.adisa.diningplus.adapters.FollowDialogAdapter
import java.util.*

class FollowDialogPreference(context: Context?, attrs: AttributeSet?) : DialogPreference(context, attrs) {
    public override fun onBindDialogView(view: View) {
        val listView = view.findViewById<View>(R.id.followList) as ListView
        val preferences = PreferenceManager.getDefaultSharedPreferences(context)
        val stringSet = preferences.getStringSet("followedItems", HashSet()) as HashSet<String>?
        val followDialogAdapter = FollowDialogAdapter(context, stringSet)
        listView.adapter = followDialogAdapter
        super.onBindDialogView(view)
    }

    override fun onPrepareDialogBuilder(builder: AlertDialog.Builder) {
        builder.setNegativeButton(null, null)
        super.onPrepareDialogBuilder(builder)
    }

    override fun onDialogClosed(positiveResult: Boolean) {
        super.onDialogClosed(positiveResult)
        persistBoolean(positiveResult)
    }

    init {
        dialogLayoutResource = R.layout.follow_dialog
    }
}